File: C-Sharp/modRecordSet.cs
using CrystalDecisions.CrystalReports.Engine;
using CrystalDecisions.Windows.Forms;
using DpSdkEngLib;
using DPSDKOPSLib;
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Xml.Linq;
 // ERROR: Not supported in C#: OptionDeclaration
namespace _4PosBackOffice.NET
{
	static class modRecordSet
	{
		public static ADODB.Connection cnnDB;
		public static ADODB.Connection cnnDB1;
			//2 Open Instances......
		public static ADODB.Connection cnnIns;
		public static bool gDisp;
		public static bool picP;
		public static bool gDisplay;
		public static short gDQuote;
		public static short gMain;
		public static short gInt;
		public static string StrLocRep;
		public static string RptHead;
		public static short gPersonID;
		public static string serverPath;
		public static short dltDayEndID;
		public static short intCur;
		public static string[] intArrayProd = new string[101];
		public static short[] intArrayCode = new short[101];
		public static short[] intArrayQty = new short[101];
//loading languages files
		public static ADODB.Recordset rsLang;
		public static short iLangRTL;
//loading help file
		public static ADODB.Recordset rsHelp;

		public static void setServerPath()
		{
			return;
			Scripting.FileSystemObject FSO = new Scripting.FileSystemObject();
			int hkey = 0;
			int lRetVal = 0;
			string vValue = null;
			string lPath = null;
			lRetVal = modUtilities.RegOpenKeyEx(modUtilities.HKEY_LOCAL_MACHINE, "Software\\4POS", 0, modUtilities.KEY_QUERY_VALUE, ref hkey);
			lRetVal = modUtilities.QueryValueEx(hkey, "server", ref vValue);
			modUtilities.RegCloseKey(hkey);
			if (string.IsNullOrEmpty(vValue)) {
				lPath = serverPath + "";
			} else {
				lPath = vValue;
			}

			if (Strings.Right(lPath, 1) != "\\")
				lPath = lPath + "\\";
			if (FSO.FolderExists(lPath)) {
				serverPath = lPath;
			} else {
				serverPath = "";
			}
		}

		public static void setNewServerPath()
		{
			Scripting.FileSystemObject FSO = new Scripting.FileSystemObject();
			int hkey = 0;
			int lRetVal = 0;
			string vValue = null;
			string lPath = null;
			lRetVal = modUtilities.RegOpenKeyEx(modUtilities.HKEY_LOCAL_MACHINE, "Software\\4POS", 0, modUtilities.KEY_QUERY_VALUE, ref hkey);
			lRetVal = modUtilities.QueryValueEx(hkey, "server", ref vValue);
			modUtilities.RegCloseKey(hkey);
			if (string.IsNullOrEmpty(vValue)) {
				lPath = serverPath + "";
			} else {
				lPath = vValue;
			}

			if (Strings.Right(lPath, 1) != "\\")
				lPath = lPath + "\\";
			if (FSO.FolderExists(lPath)) {
				serverPath = lPath;
			} else {
				serverPath = "";
			}
		}

		public static ADODB.Recordset getRS(ref string sql)
		{
			ADODB.Recordset functionReturnValue = default(ADODB.Recordset);
			functionReturnValue = new ADODB.Recordset();
			functionReturnValue.CursorLocation = ADODB.CursorLocationEnum.adUseClient;
			functionReturnValue.Open(sql, cnnDB, ADODB.CursorTypeEnum.adOpenStatic, ADODB.LockTypeEnum.adLockOptimistic, -1);
			return functionReturnValue;
			//Debug.Print(sql)
		}
		public static bool openConnection()
		{
			bool functionReturnValue = false;
			string Path = null;
			string cnnDBname = null;
			//this
			 // ERROR: Not supported in C#: OnErrorStatement

			//If Win7Ver() = True Then
			//Path = "c:\4posserver\"
			//Else
			//Path = String.Empty
			//End If
			cnnDBname = "4pos";
			cnnDB = new ADODB.Connection();
			//cnnDB.Provider = "Microsoft.ACE.OLEDB.12.0"
			cnnDB.Open(cnnDBname);

			//win 7
			Scripting.FileSystemObject FSO = new Scripting.FileSystemObject();
			Scripting.TextStream textstream = default(Scripting.TextStream);
			string lString = null;
			if (modWinVer.Win7Ver() == true) {
				if (FSO.FileExists("C:\\4POS\\4POSWinPath.txt")) {
					textstream = FSO.OpenTextFile("C:\\4POS\\4POSWinPath.txt", Scripting.IOMode.ForReading, true);
					lString = textstream.ReadAll;
					serverPath = lString;
					//& "pricing.mdb"
				} else {
					serverPath = "C:\\4POSServer\\";
					//"pricing.mdb"
				}
				functionReturnValue = true;
				return functionReturnValue;
			}
			//win 7

			serverPath = Strings.Split(Strings.Split(cnnDB.ConnectionString, ";DBQ=")[1], ";")[0];
			serverPath = Strings.Split(Strings.LCase(serverPath), "pricing.mdb")[0];
			//
			functionReturnValue = true;
			return functionReturnValue;
			if (string.IsNullOrEmpty(serverPath))
				setServerPath();
			string strDBPath = null;
			if (string.IsNullOrEmpty(serverPath)) {
				functionReturnValue = false;
			} else {
				functionReturnValue = true;
				cnnDB = new ADODB.Connection();
				strDBPath = serverPath + "Pricing.mdb";
				var _with1 = cnnDB;
				_with1.Provider = "Microsoft.ACE.OLEDB.12.0";
				_with1.Properties("Jet OLEDB:System Database").Value = serverPath + "secured.mdw";
				_with1.Open(strDBPath, "liquid", "lqd");
			}
			return functionReturnValue;
			withPass:

			if (string.IsNullOrEmpty(serverPath))
				setNewServerPath();
			if (!string.IsNullOrEmpty(serverPath)) {
				cnnDB = new ADODB.Connection();
				strDBPath = serverPath + "pricing.mdb";
				Path = strDBPath + ";Jet " + "OLEDB:Database Password=lqd";
				cnnDB.CursorLocation = ADODB.CursorLocationEnum.adUseClient;
				cnnDB.Open("Provider=Microsoft.ACE.OLEDB.12.0;Persist Security Info=False;Data Source= " + Path);
				functionReturnValue = true;
			}
			return functionReturnValue;
			openConnection_Error:

			if (Err().Description == "[Microsoft][ODBC Microsoft Access Driver] Not a valid password.") {
				goto withPass;
			} else if (Err().Description == "Not a valid password.") {
				goto withPass;
			}

			functionReturnValue = false;
			return functionReturnValue;

		}
		public static ADODB.Connection openConnectionInstance(ref string lDatabase = "")
		{
			ADODB.Connection functionReturnValue = default(ADODB.Connection);
			string Path = null;
			Scripting.FileSystemObject FSO = new Scripting.FileSystemObject();

			if (string.IsNullOrEmpty(lDatabase))
				lDatabase = "Pricing.mdb";
			if (string.IsNullOrEmpty(serverPath))
				setServerPath();
			 // ERROR: Not supported in C#: OnErrorStatement

			ADODB.Connection cn = new ADODB.Connection();
			//UPGRADE_NOTE: Object openConnectionInstance may not be destroyed until it is garbage collected. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6E35BFF6-CD74-4B09-9689-3E1A43DF8969"'
			functionReturnValue = null;
			string strDBPath = null;
			if (string.IsNullOrEmpty(serverPath)) {
			} else {
				strDBPath = serverPath + lDatabase;
				var _with2 = cn;
				_with2.Provider = "Microsoft.ACE.OLEDB.12.0";
				_with2.Properties("Jet OLEDB:System Database").Value = serverPath + "Secured.mdw";
				_with2.Open(strDBPath, "liquid", "lqd");
				functionReturnValue = cn;
			}
			return functionReturnValue;
			withPass:

			if (string.IsNullOrEmpty(serverPath))
				setNewServerPath();
			if (!string.IsNullOrEmpty(serverPath)) {
				functionReturnValue = null;
				strDBPath = serverPath + lDatabase;
				Path = strDBPath + ";Jet " + "OLEDB:Database Password=lqd";
				cn.CursorLocation = ADODB.CursorLocationEnum.adUseClient;
				cn.Open("Provider=Microsoft.ACE.OLEDB.12.0;Persist Security Info=False;Data Source= " + Path);
				functionReturnValue = cn;
			}
			return functionReturnValue;
			openConnection_Error:

			if (Err().Description == "[Microsoft][ODBC Microsoft Access Driver] Not a valid password.") {
				goto withPass;
			} else if (Err().Description == "Not a valid password.") {
				goto withPass;
			} else {
				Interaction.MsgBox(Err().Number + " - " + Err().Description);
			}
			return functionReturnValue;

		}
		public static ADODB.Connection openConnectionShapeInstance()
		{
			ADODB.Connection functionReturnValue = default(ADODB.Connection);
			string Path = null;
			string lDatabase = null;
			 // ERROR: Not supported in C#: OnErrorStatement

			ADODB.Connection cn = new ADODB.Connection();
			functionReturnValue = null;
			if (string.IsNullOrEmpty(serverPath))
				setServerPath();
			string strDBPath = null;
			if (string.IsNullOrEmpty(serverPath)) {
			} else {
				strDBPath = serverPath + "Pricing.mdb";
				var _with3 = cn;
				_with3.Provider = "MSDataShape";
				cn.Open("Data Provider=Microsoft.ACE.OLEDB.12.0;Data Source=" + strDBPath + ";User Id=liquid;Password=lqd;Jet OLEDB:System Database=" + serverPath + "Secured.mdw");
				functionReturnValue = cn;
			}
			return functionReturnValue;
			withPass:

			if (string.IsNullOrEmpty(serverPath))
				setNewServerPath();
			if (!string.IsNullOrEmpty(serverPath)) {
				functionReturnValue = null;
				strDBPath = serverPath + lDatabase;
				Path = strDBPath + ";Jet " + "OLEDB:Database Password=lqd";
				cn.CursorLocation = ADODB.CursorLocationEnum.adUseClient;
				cn.Open("Provider=Microsoft.ACE.OLEDB.12.0;Persist Security Info=False;Data Source= " + Path);
				functionReturnValue = cn;
			}
			return functionReturnValue;
			openConnectionShape_Error:

			if (Err().Description == "[Microsoft][ODBC Microsoft Access Driver] Not a valid password.") {
				goto withPass;
			} else if (Err().Description == "Not a valid password.") {
				goto withPass;
			} else {
				Interaction.MsgBox(Err().Number + " - " + Err().Description);
			}
			return functionReturnValue;
		}
		public static void closeConnection()
		{
			 // ERROR: Not supported in C#: OnErrorStatement

			cnnDB.Close();
			cnnDB = null;
		}


		public static ADODB.Recordset getRSwaitron(ref object sql, ref ADODB.Connection cn)
		{
			ADODB.Recordset functionReturnValue = default(ADODB.Recordset);
			functionReturnValue = new ADODB.Recordset();
			functionReturnValue.CursorLocation = ADODB.CursorLocationEnum.adUseClient;
			functionReturnValue.Open(sql, cn, ADODB.CursorTypeEnum.adOpenStatic, ADODB.LockTypeEnum.adLockOptimistic);
			return functionReturnValue;

		}


		public static ADODB.Connection openSComp(ref string lDatabase = "")
		{
			ADODB.Connection functionReturnValue = default(ADODB.Connection);
			Scripting.FileSystemObject FSO = new Scripting.FileSystemObject();

			//If lDatabase = "" Then lDatabase = "Pricing.mdb"
			//If serverPath = "" Then setServerPath
			if (string.IsNullOrEmpty(lDatabase))
				lDatabase = serverPath + "Pricing.mdb";
			 // ERROR: Not supported in C#: OnErrorStatement

			ADODB.Connection cn = new ADODB.Connection();
			functionReturnValue = null;
			string strDBPath = null;
			if (string.IsNullOrEmpty(lDatabase)) {
			} else {
				strDBPath = lDatabase;
				var _with4 = cn;
				_with4.Provider = "Microsoft.ACE.OLEDB.12.0";
				_with4.Properties("Jet OLEDB:System Database").Value = serverPath + "Secured.mdw";
				_with4.Open(strDBPath, "liquid", "lqd");
				functionReturnValue = cn;
			}
			return functionReturnValue;
			openConnection_Error:
			return functionReturnValue;


		}
	}
}

File: VB/4PosBackOffice.NET/modRecordSet.vb
Option Strict Off
Option Explicit On
Module modRecordSet
	Public cnnDB As ADODB.Connection
	Public cnnDB1 As ADODB.Connection
	Public cnnIns As ADODB.Connection '2 Open Instances......
	Public gDisp As Boolean
	Public picP As Boolean
	Public gDisplay As Boolean
	Public gDQuote As Short
	Public gMain As Short
	Public gInt As Short
	Public StrLocRep As String
	Public RptHead As String
	Public gPersonID As Short
	Public serverPath As String
	Public dltDayEndID As Short
	Public intCur As Short
	Public intArrayProd(100) As String
	Public intArrayCode(100) As Short
	Public intArrayQty(100) As Short
	'loading languages files
	Public rsLang As ADODB.Recordset
	Public iLangRTL As Short
	'loading help file
	Public rsHelp As ADODB.Recordset
	
	Public Sub setServerPath()
		Exit Sub
		Dim FSO As New Scripting.FileSystemObject
        Dim hkey As Integer
		Dim lRetVal As Integer
		Dim vValue As String
		Dim lPath As String
		lRetVal = RegOpenKeyEx(HKEY_LOCAL_MACHINE, "Software\4POS", 0, KEY_QUERY_VALUE, hkey)
		lRetVal = QueryValueEx(hkey, "server", vValue)
		RegCloseKey(hkey)
		If vValue = "" Then
			lPath = serverPath & ""
		Else
			lPath = vValue
		End If
		
		If Right(lPath, 1) <> "\" Then lPath = lPath & "\"
		If FSO.FolderExists(lPath) Then
			serverPath = lPath
		Else
			serverPath = ""
		End If
	End Sub
	
	Public Sub setNewServerPath()
		Dim FSO As New Scripting.FileSystemObject
        Dim hkey As Integer
		Dim lRetVal As Integer
		Dim vValue As String
		Dim lPath As String
		lRetVal = RegOpenKeyEx(HKEY_LOCAL_MACHINE, "Software\4POS", 0, KEY_QUERY_VALUE, hkey)
		lRetVal = QueryValueEx(hkey, "server", vValue)
		RegCloseKey(hkey)
		If vValue = "" Then
			lPath = serverPath & ""
		Else
			lPath = vValue
		End If
		
		If Right(lPath, 1) <> "\" Then lPath = lPath & "\"
		If FSO.FolderExists(lPath) Then
			serverPath = lPath
		Else
			serverPath = ""
		End If
	End Sub
	
    Public Function getRS(ByRef sql As String) As ADODB.Recordset
        getRS = New ADODB.Recordset
        getRS.CursorLocation = ADODB.CursorLocationEnum.adUseClient
        getRS.Open(sql, cnnDB, ADODB.CursorTypeEnum.adOpenStatic, ADODB.LockTypeEnum.adLockOptimistic, -1)
        'Debug.Print(sql)
    End Function
	Public Function openConnection() As Boolean
        Dim Path As String
        Dim cnnDBname As String
		'this
        On Error GoTo openConnection_Error
        'If Win7Ver() = True Then
        'Path = "c:\4posserver\"
        'Else
        'Path = String.Empty
        'End If
        cnnDBname = "4pos"
        cnnDB = New ADODB.Connection
        'cnnDB.Provider = "Microsoft.ACE.OLEDB.12.0"
        cnnDB.Open(cnnDBname)

        'win 7
        Dim FSO As New Scripting.FileSystemObject
        Dim textstream As Scripting.TextStream
        Dim lString As String
        If Win7Ver() = True Then
            If FSO.FileExists("C:\4POS\4POSWinPath.txt") Then
                textstream = FSO.OpenTextFile("C:\4POS\4POSWinPath.txt", Scripting.IOMode.ForReading, True)
                lString = textstream.ReadAll
                serverPath = lString '& "pricing.mdb"
            Else
                serverPath = "C:\4POSServer\" '"pricing.mdb"
            End If
            openConnection = True
            Exit Function
        End If
        'win 7

        serverPath = Split(Split(cnnDB.ConnectionString, ";DBQ=")(1), ";")(0)
        serverPath = Split(LCase(serverPath), "pricing.mdb")(0) '
        openConnection = True
        Exit Function
        If serverPath = "" Then setServerPath()
        Dim strDBPath As String
        If serverPath = "" Then
            openConnection = False
        Else
            openConnection = True
            cnnDB = New ADODB.Connection
            strDBPath = serverPath & "Pricing.mdb"
            With cnnDB
                .Provider = "Microsoft.ACE.OLEDB.12.0"
                .Properties("Jet OLEDB:System Database").Value = serverPath & "secured.mdw"
                .Open(strDBPath, "liquid", "lqd")
            End With
        End If
        Exit Function

withPass:
        If serverPath = "" Then setNewServerPath()
        If serverPath <> "" Then
            cnnDB = New ADODB.Connection
            strDBPath = serverPath & "pricing.mdb"
            Path = strDBPath & ";Jet " & "OLEDB:Database Password=lqd"
            cnnDB.CursorLocation = ADODB.CursorLocationEnum.adUseClient
            cnnDB.Open("Provider=Microsoft.ACE.OLEDB.12.0;Persist Security Info=False;Data Source= " & Path)
            openConnection = True
        End If
        Exit Function

openConnection_Error:
        If Err.Description = "[Microsoft][ODBC Microsoft Access Driver] Not a valid password." Then
            GoTo withPass
        ElseIf Err.Description = "Not a valid password." Then
            GoTo withPass
        End If

        openConnection = False

	End Function
	Public Function openConnectionInstance(Optional ByRef lDatabase As String = "") As ADODB.Connection
        Dim Path As String
		Dim FSO As New Scripting.FileSystemObject
		
		If lDatabase = "" Then lDatabase = "Pricing.mdb"
		If serverPath = "" Then setServerPath()
		On Error GoTo openConnection_Error
		Dim cn As New ADODB.Connection
		'UPGRADE_NOTE: Object openConnectionInstance may not be destroyed until it is garbage collected. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6E35BFF6-CD74-4B09-9689-3E1A43DF8969"'
		openConnectionInstance = Nothing
		Dim strDBPath As String
		If serverPath = "" Then
		Else
			strDBPath = serverPath & lDatabase
			With cn
				.Provider = "Microsoft.ACE.OLEDB.12.0"
				.Properties("Jet OLEDB:System Database").Value = serverPath & "Secured.mdw"
				.Open(strDBPath, "liquid", "lqd")
			End With
			openConnectionInstance = cn
		End If
		Exit Function
		
withPass: 
		If serverPath = "" Then setNewServerPath()
		If serverPath <> "" Then
			openConnectionInstance = Nothing
			strDBPath = serverPath & lDatabase
			Path = strDBPath & ";Jet " & "OLEDB:Database Password=lqd"
			cn.CursorLocation = ADODB.CursorLocationEnum.adUseClient
			cn.Open("Provider=Microsoft.ACE.OLEDB.12.0;Persist Security Info=False;Data Source= " & Path)
			openConnectionInstance = cn
		End If
		Exit Function
		
openConnection_Error: 
		If Err.Description = "[Microsoft][ODBC Microsoft Access Driver] Not a valid password." Then
			GoTo withPass
		ElseIf Err.Description = "Not a valid password." Then 
			GoTo withPass
		Else
			MsgBox(Err.Number & " - " & Err.Description)
		End If
		
	End Function
	Public Function openConnectionShapeInstance() As ADODB.Connection
        Dim Path As String
        Dim lDatabase As String
		On Error GoTo openConnectionShape_Error
		Dim cn As New ADODB.Connection
		openConnectionShapeInstance = Nothing
		If serverPath = "" Then setServerPath()
		Dim strDBPath As String
		If serverPath = "" Then
		Else
			strDBPath = serverPath & "Pricing.mdb"
			With cn
				.Provider = "MSDataShape"
				cn.Open("Data Provider=Microsoft.ACE.OLEDB.12.0;Data Source=" & strDBPath & ";User Id=liquid;Password=lqd;Jet OLEDB:System Database=" & serverPath & "Secured.mdw")
			End With
			openConnectionShapeInstance = cn
		End If
		Exit Function
		
withPass: 
		If serverPath = "" Then setNewServerPath()
		If serverPath <> "" Then
			openConnectionShapeInstance = Nothing
			strDBPath = serverPath & lDatabase
			Path = strDBPath & ";Jet " & "OLEDB:Database Password=lqd"
			cn.CursorLocation = ADODB.CursorLocationEnum.adUseClient
			cn.Open("Provider=Microsoft.ACE.OLEDB.12.0;Persist Security Info=False;Data Source= " & Path)
			openConnectionShapeInstance = cn
		End If
		Exit Function
		
openConnectionShape_Error: 
		If Err.Description = "[Microsoft][ODBC Microsoft Access Driver] Not a valid password." Then
			GoTo withPass
		ElseIf Err.Description = "Not a valid password." Then 
			GoTo withPass
		Else
			MsgBox(Err.Number & " - " & Err.Description)
		End If
	End Function
	Public Sub closeConnection()
		On Error Resume Next
		cnnDB.Close()
		cnnDB = Nothing
	End Sub
	
	
	Public Function getRSwaitron(ByRef sql As Object, ByRef cn As ADODB.Connection) As ADODB.Recordset
		getRSwaitron = New ADODB.Recordset
		getRSwaitron.CursorLocation = ADODB.CursorLocationEnum.adUseClient
		getRSwaitron.Open(sql, cn, ADODB.CursorTypeEnum.adOpenStatic, ADODB.LockTypeEnum.adLockOptimistic)
		
	End Function
	
	
	Public Function openSComp(Optional ByRef lDatabase As String = "") As ADODB.Connection
		Dim FSO As New Scripting.FileSystemObject
		
		'If lDatabase = "" Then lDatabase = "Pricing.mdb"
		'If serverPath = "" Then setServerPath
		If lDatabase = "" Then lDatabase = serverPath & "Pricing.mdb"
		On Error GoTo openConnection_Error
		Dim cn As New ADODB.Connection
		openSComp = Nothing
		Dim strDBPath As String
		If lDatabase = "" Then
		Else
			strDBPath = lDatabase
			With cn
				.Provider = "Microsoft.ACE.OLEDB.12.0"
				.Properties("Jet OLEDB:System Database").Value = serverPath & "Secured.mdw"
				.Open(strDBPath, "liquid", "lqd")
			End With
			openSComp = cn
		End If
		
		Exit Function
openConnection_Error: 
		
	End Function
End Module